app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java

```diff
@@ -0,0 +1,152 @@
+
+    GoogleMap mGoogleMap;
+    MapView mMapView;
+    private EditText mSearchText;
+    private static final String TAG = "Location";
+    private Button button;
+    LatLng latLng;
+
+
+    private void inity() {
+        mGoogleMap.setOnMapLongClickListener(new GoogleMap.OnMapLongClickListener(){
+
+            @Override
+            public void onMapLongClick(LatLng Longl) {
+                latLng = Longl;
+                setMarker(latLng);
+
+            }
+        });
+        mSearchText = (EditText) findViewById(R.id.input_search);
+        mSearchText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView textView, int actionId, KeyEvent event) {
+                if (actionId == EditorInfo.IME_ACTION_SEARCH
+                        || actionId == EditorInfo.IME_ACTION_DONE
+                        || event.getAction() == KeyEvent.ACTION_DOWN
+                        || event.getAction() == KeyEvent.KEYCODE_ENTER) {
+                    geoLocate();
+                }
+                return false;
+            }
+        });
+    }
+
+    private void geoLocate() {
+        String searchingString = mSearchText.getText().toString();
+        Geocoder geocoder = new Geocoder(this);
+        List<Address> list = new ArrayList<>();
+        try {
+            list = geocoder.getFromLocationName(searchingString, 1);
+
+
+        } catch (IOException e) {
+        }
+        if (list.size() > 0) {
+            Address address = list.get(0);
+            Log.d(TAG, "Locate adress:" + address.toString());
+            //Toast.makeText(this, address.toString(), Toast.LENGTH_SHORT).show();
+            CameraPosition search = CameraPosition.builder().target(new LatLng(address.getLatitude(),
+                    address.getLongitude())).zoom(10).bearing(0).tilt(0).build();
+            mGoogleMap.moveCamera(CameraUpdateFactory.newCameraPosition(search));
+            latLng =  new LatLng(address.getLatitude(), address.getLongitude());
+            setMarker(latLng);
+
+
+
+
+        }
+    }
+    private void setMarker(LatLng latLng){
+        mGoogleMap.clear();;
+        MarkerOptions options = new MarkerOptions().position(latLng);
+        mGoogleMap.addMarker(options);
+
+    }
+
+
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_location_edit);
+        button = findViewById(R.id.memory_save_location_button);
+        button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent returnIntent = new Intent();
+                returnIntent.putExtra("result",latLng);
+                setResult(Activity.RESULT_OK,returnIntent);
+                finish();
+
+            }
+        });
+        mMapView = findViewById(R.id.map);
+        if (mMapView != null) {
```

Comment:
mMapView is never null. This `if` statement is useless.

Reply:
How do you know?

Reply:
done

Reply:
Because `mMapView` is set in the line above it.